src/main/java/com/google/devtools/build/lib/skyframe/RegisteredExecutionPlatformsFunction.java

```diff
@@ -109,27 +113,55 @@ public SkyValue compute(SkyKey skyKey, Environment env)
   }
 
   private ImmutableList<ConfiguredTargetKey> configureRegisteredExecutionPlatforms(
-      Environment env, BuildConfiguration configuration, List<Label> labels)
+      Environment env,
+      BuildConfiguration configuration,
+      ImmutableList<String> targetPatterns,
+      List<Label> labels)
       throws InterruptedException, RegisteredExecutionPlatformsFunctionException {
+
     ImmutableList<ConfiguredTargetKey> keys =
-        labels
-            .stream()
+        labels.stream()
             .map(label -> ConfiguredTargetKey.of(label, configuration))
-            .collect(ImmutableList.toImmutableList());
-
-    // Load the actual configured targets and ensure that they have real, valid PlatformInfo
-    // instances. These are loaded later during toolchain resolution, so this is work that needs to
-    // be done anyway, but here we can fail fast on an error.
-    try {
-      PlatformLookupUtil.getPlatformInfo(keys, env);
-      if (env.valuesMissing()) {
-        return null;
+            .collect(toImmutableList());
+
+    Map<SkyKey, ValueOrException<ConfiguredValueCreationException>> values =
+        env.getValuesOrThrow(keys, ConfiguredValueCreationException.class);
+    ImmutableList.Builder<ConfiguredTargetKey> validPlatformKeys = new ImmutableList.Builder<>();
+    boolean valuesMissing = false;
+    for (ConfiguredTargetKey platformKey : keys) {
+      Label platformLabel = platformKey.getLabel();
+      try {
+        ValueOrException<ConfiguredValueCreationException> valueOrException =
+            values.get(platformKey);
+        if (valueOrException.get() == null) {
+          valuesMissing = true;
+          continue;
+        }
+        ConfiguredTarget target =
+            ((ConfiguredTargetValue) valueOrException.get()).getConfiguredTarget();
+        PlatformInfo platformInfo = PlatformProviderUtils.platform(target);
+        if (platformInfo == null) {
+          if (TargetPatternUtil.isTargetExplicit(targetPatterns, platformLabel)) {
+            // Only report an error if the label was explicitly requested.
+            throw new RegisteredExecutionPlatformsFunctionException(
+                new InvalidPlatformException(platformLabel), Transience.PERSISTENT);
```

Comment:
Yes, but. getPlatformInfo returns the actual map of platform providers. This version implements the same lookup logic, but instead returns a list of ConfiguredTargetKeys that correspond to targets that have a platform provider. It may also emit an error if the configured target was directly requested (ie, isn't part of a target pattern expansion).